hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/TransactionHandler.java

```diff
@@ -33,6 +34,18 @@ public interface TransactionHandler {
      */
     void preHandle(@NonNull final PreHandleContext context) throws PreCheckException;
 
+    /**
+     * Validate the transaction body, without involving state or dynamic properties.
+     * This method is called as first step of preHandle. If there is any failure,
+     * throws a {@link PreCheckException}.
+     * Since these checks are pure, they need not be repeated in handle workflow.
+     * The result of these checks is cached in the {@link PreHandleContext} for use
+     * in handle workflow.
+     * @param txn the transaction body
+     * @throws PreCheckException if the transaction is invalid
+     */
```

Comment:
I had thought of generifying the method signature so we can pass in each operation specifically instead of passing in `TransactionBody` everywhere:

```java
public interface TransactionHandler<T> {
    void pureChecks(@NonNull final T operation) throws PreCheckException;
}

// would result in an implementation similar to the following:
public class MyFakeOperation implements TransactionHandler<MyFakeOperation> {
    @override
    void pureChecks (@NonNull final MyFakeOperation op) {
        // checks here
    }
}
```
Thoughts?

Reply:
This would likely add complexity to the workflows, which would now have to do the conversion to the specific subtype of transaction body, which I believe is something we've been avoiding.
The subtypes are not subclasses of TransactionBody because of how Protobuf types work, so it's not a simple drop-in change.